docs/event/assets.md

```diff
@@ -0,0 +1,101 @@
+---
+sidebar_position: 4
+---
+
+# Assets
+
+Assets are files associated with an event.
+Each asset has some asset-metadata that is stored in the DB, while the file itself is stored in the file system and is referenced by the database.
+
+Every asset has this metadata attached to it:
+- `id: ID`: unique identifier among all assets of all events. Assigned by Opencast and unchangable.
+- `flavor: NonBlankAsciiString` <sup>(1?)</sup>
+- `tags: string[]` <sup>(1?)</sup>
+- `properties: Map<Label, string>`: a `Label` to string map for custom properties. Values can be arbitrary strings.<sup>(7?)</sup>
+- `mimeType: NonBlankAsciiString`: a *lowercase* `NonBlankAsciiString` representing the MIME-type of the asset.
+- `size: uint64`: size of the file in bytes. This is always the actual file size and cannot be changed manually.
+- `checksum`: Checksum/hash of the file. Consists of `type` (e.g. `md5`, `sha256`) and the hex encoded value. In the API (and maybe in the database?) this should be serialized as `<type>:<value>`, e.g. `sha256:e3b0c44298f...`.
+- `source: bool`: whether this was directly uploaded by a user. `false` if it was processed or generated by Opencast.
+- `updated: Timestamp`: timestamp of when anything about this asset was last changed. <sup>(2?)</sup>
+- `internal: bool`: internal assets are not exposed like other assets in APIs. They are used to store source or intermediate artifacts, like the originally uploaded video. These cannot be queried or read by users with only `read` access. <sup>(3?)</sup>
+
+Additionally, in the API representation, assets have the following fields:
+- `uri: string`: a URI to the asset, i.e. where it can be downloaded.
+
+
+## Tracks
+
+A track is an attachment with time-component and "lives on a timeline", i.e. representing something that spans the video length.
+A cut operation on the video needs to modify all tracks.
+
+Each track has one or more streams.
+For example, an `mp4` track might have a video and audio stream, while a `vtt` track only has a single text track.
+
+- `tracks: Track[]`
+- `isLive: bool` <sup>(5?)</sup>
+- `isMaster: bool`: <sup>TODO: describe what exactly this means</sup>
+- `duration: Milliseconds`
+
+Non-`internal` tracks have to have the following properties:
+- The `duration` of the track has to match the `duration` of the event, i.e. all non-internal durations are the same.
+- There is only one video stream<sup>?</sup>
+
+
+### Streams
+
+A stream has the following properties:
+- `language: LangCode?`
+- `type: "video" | "audio" | "text"`: type of the track. Depending on the type, there are additional properties:
+  - `"video"`
+    - `resolution: [uint32, uint32]`
+    - `framerate: "dynamic" | float32`<sup>?</sup>
```

Comment:
`average framerate`, `lowest common framerate` (ffprobe is reporting this, not sure how to actually call this).